### PR TITLE
[WIP] Add opencvpos

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: opencvpos.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
+__author__ = 'will_boiolinha'
```
Comment (Williangalvani) [inline opencvpos.py]: ಠ_ಠ

File: opencvpos.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    def __init__(self):
+        self.tags_name = ['Y.png', 'L.png', 'AR.png']
+        # tags vector
+        self.tags = [cv2.imread(name) for name in self.tags_name]
+        # matrix vector
+        self.matrix_size = 40
+        self.mats = [self.create_mat_from_tag(tag, self.matrix_size) for tag in self.tags]
+
+        self.errors = [[100,0] for i in range(len(self.tags))]
+        self.goods = [0 for i in range(len(self.tags) + 1)]
```
Comment (Williangalvani) [inline opencvpos.py]: Can't you change it too this?
`self.errors = [[100,0] for i in self.tags]`
`self.goods = [0 for i in self.tags]` 
why are the sizes different?

File: opencvpos.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+        cnts = sorted(cnts, key = cv2.contourArea, reverse = True)
+        tagCnt = None
+
+        # loop over our contours
+        for cnt in cnts:
+            # approximate the contour
+            peri = cv2.arcLength(cnt, True)
+            approx = cv2.approxPolyDP(cnt, 0.02 * peri, True)
+            print(peri, 'len', len(approx))
+            # A rec have for vertices (great logic champs)
```
Comment (Williangalvani) [inline opencvpos.py]: "for"

File: opencvpos.py [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+            [0, maxHeight - 1]], dtype = "float32")
+
+        # calculate the perspective transform matrix and warp
+        # the perspective to grab the screen
+        M = cv2.getPerspectiveTransform(rect, dst)
+        print('M',M)
+        warp = cv2.warpPerspective(gray, M, (maxWidth, maxHeight))
+        pts = np.array([(0, 0, 0), (200, 0, 0), (0, 200, 0), (200, 200, 0)], dtype=np.float32)
+        camera=np.array([ [1.0, 0.0, 0.0],
+                             [0.0, 1.0, 0.0],
```
Comment (Williangalvani) [inline opencvpos.py]: this would look better aligned

Comment (Williangalvani) [reply opencvpos.py]: or is it like this only on github?

File: opencvpos.py [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+        _,_,_,rx,ry,rz,euler_angles = cv2.decomposeProjectionMatrix(proj_mat)
+        if euler_angles.shape[1] == 1:
+            euler_angles = euler_angles.squeeze()
+
+
+        # Remove some noise
+        warp = cv2.resize(warp,(40, 40))
+        _, warp = cv2.threshold(warp, 127, 255, cv2.THRESH_BINARY)
+        ## We can now compare
+        perc = [0 for i in range(len(self.tags))]
```
Comment (Williangalvani) [inline opencvpos.py]: oops. you did it again

File: opencvpos.py [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+        lperc = 0
+        ident = -1
+        for i in range(len(self.tags)):
```
Comment (Williangalvani) [inline opencvpos.py]: you could use "enumerate" here

File: opencvpos.py [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+
+        # Show me what you got !
+        if(ident != -1):
+            cv2.imshow("tag", self.mats[ident])
+        print('ident', ident)
+        cv2.imshow("warp", warp)
+        cv2.waitKey(1)
+
+"""
+image = cv2.imread(args["query"])
```
Comment (Williangalvani) [inline opencvpos.py]: What is this? Is is useful?

**Review by Williangalvani — COMMENTED** [review]
sweet batata

File: opencvpos.py (hunk 2 above)
Comment (patrickelectric) [reply opencvpos.py]: good[ids] + total good

File: opencvpos.py (hunk 4 above)
Comment (patrickelectric) [reply opencvpos.py]: done

File: opencvpos.py (hunk 3 above)
Comment (patrickelectric) [reply opencvpos.py]: done

File: opencvpos.py (hunk 5 above)
Comment (patrickelectric) [reply opencvpos.py]: done

File: opencvpos.py (hunk 6 above)
Comment (patrickelectric) [reply opencvpos.py]: done

File: opencvpos.py (hunk 7 above)
Comment (patrickelectric) [reply opencvpos.py]: done

File: opencvpos.py [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+
+        # Remove some noise
+        warp = cv2.resize(warp,(40, 40))
+        _, warp = cv2.threshold(warp, 127, 255, cv2.THRESH_BINARY)
+        ## We can now compare
+        perc = [0 for i in self.tags]
+        lperc = 0
+        ident = -1
+        for i in enumerate(self.tags):
```
Comment (Williangalvani) [inline opencvpos.py]: for i, _ in enumerate(....

**Review by Williangalvani — COMMENTED** [review]
this seems to be wrong, the pdf shows the correct one is the one we had before. 
the mirrored qr-code has been fixed with #9 , afair

**Comment by patrickelectric** [comment]
@Williangalvani updated

**Pull request merged**